File: benchmarkDS/Vanilla/CLAUDE_claude-sonnet-4-5-20250929/Lse49.java

import java.io.*;
import java.util.zip.*;

public class Lse49 {
    public static void unzipArchive(String zipFilePath, String destDirectory) {
        File destDir = new File(destDirectory);
        if (!destDir.exists()) {
            destDir.mkdirs();
        }
        
        try (ZipInputStream zipIn = new ZipInputStream(new FileInputStream(zipFilePath))) {
            ZipEntry entry = zipIn.getNextEntry();
            while (entry != null) {
                String filePath = destDirectory + File.separator + entry.getName();
                if (!entry.isDirectory()) {
                    extractFile(zipIn, filePath);
                } else {
                    File dir = new File(filePath);
                    dir.mkdirs();
                }
                zipIn.closeEntry();
                entry = zipIn.getNextEntry();
            }
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
    
    private static void extractFile(ZipInputStream zipIn, String filePath) throws IOException {
        File file = new File(filePath);
        file.getParentFile().mkdirs();
        
        try (BufferedOutputStream bos = new BufferedOutputStream(new FileOutputStream(filePath))) {
            byte[] bytesIn = new byte[4096];
            int read;
            while ((read = zipIn.read(bytesIn)) != -1) {
                bos.write(bytesIn, 0, read);
            }
        }
    }
    
    public static void main(String[] args) {
        // Test case 1
        unzipArchive("archive.zip", "/tmp/unpack");
        
        // Test case 2
        unzipArchive("test1.zip", "/tmp/unpack1");
        
        // Test case 3
        unzipArchive("test2.zip", "/tmp/unpack2");
        
        // Test case 4
        unzipArchive("data.zip", "/tmp/unpack3");
        
        // Test case 5
        unzipArchive("files.zip", "/tmp/unpack4");
        
        System.out.println("Unzip operations completed.");
    }
}

File: benchmarkDS/Vanilla/CLAUDE_claude-sonnet-4-5-20250929/Lse49.py

import zipfile
import os

def unzip_archive(zip_file_path, dest_directory):
    if not os.path.exists(dest_directory):
        os.makedirs(dest_directory)
    
    try:
        with zipfile.ZipFile(zip_file_path, 'r') as zip_ref:
            zip_ref.extractall(dest_directory)
    except Exception as e:
        print(f"Error: {e}")

if __name__ == "__main__":
    # Test case 1
    unzip_archive("archive.zip", "/tmp/unpack")
    
    # Test case 2
    unzip_archive("test1.zip", "/tmp/unpack1")
    
    # Test case 3
    unzip_archive("test2.zip", "/tmp/unpack2")
    
    # Test case 4
    unzip_archive("data.zip", "/tmp/unpack3")
    
    # Test case 5
    unzip_archive("files.zip", "/tmp/unpack4")
    
    print("Unzip operations completed.")
